build.sbt

```diff
@@ -0,0 +1,279 @@
+  // a circular instance, see: https://github.com/sbt/sbt/issues/1872
+  managedScalaInstance := false,
+  // this is a way to workaround issue described in https://github.com/sbt/sbt/issues/1872
+  // check it out for more details
+  scalaInstance := ScalaInstance(scalaVersion.value, appConfiguration.value.provider.scalaProvider.launcher getScala scalaVersion.value),
+  // we always assume that Java classes are standalone and do not have any dependency
+  // on Scala classes
+  compileOrder := CompileOrder.JavaThenScala,
+  javacOptions in Compile ++= Seq("-g", "-source", "1.5", "-target", "1.6"),
+  // we don't want any unmanaged jars; as a reminder: unmanaged jar is a jar stored
+  // directly on the file system and it's not resolved through Ivy
+  // Ant's build stored unmanaged jars in `lib/` directory
+  unmanagedJars in Compile := Seq.empty,
+  sourceDirectory in Compile := baseDirectory.value,
+  sourceDirectories in Compile := Seq(sourceDirectory.value),
+  scalaSource in Compile := (sourceDirectory in Compile).value,
+  javaSource in Compile := (sourceDirectory in Compile).value,
+  target := (baseDirectory in ThisBuild).value / "target" / thisProject.value.id,
+  target in Compile in doc := buildDirectory.value / "scaladoc" / thisProject.value.id,
+  classDirectory in Compile := buildDirectory.value / "quick/classes" / thisProject.value.id,
+  // given that classDirectory is overriden to be _outside_ of target directory, we have
+  // to make sure its being cleaned properly
+  cleanFiles += (classDirectory in Compile).value
+)
+
+// disable various tasks that are not needed for projects that are used
+// only for compiling code and not publishing it as a standalone artifact
+// we disable those tasks by overriding them and returning bogus files when
+// needed. This is a bit sketchy but I haven't found any better way.
+val disableDocsAndPublishingTasks = Seq[Setting[_]](
+  (doc := file("!!! NO DOCS !!!")),
+  (publishLocal := {}),
+  (publish := {}),
+  (packageBin in Compile := file("!!! NO PACKAGING !!!"))
+)
+
+lazy val scalaSubprojectSettings = commonSettings ++ Seq[Setting[_]](
+  artifactPath in packageBin in Compile := {
+    // two lines below are copied over from sbt's sources:
+    // https://github.com/sbt/sbt/blob/0.13/main/src/main/scala/sbt/Defaults.scala#L628
+    //val resolvedScalaVersion = ScalaVersion((scalaVersion in artifactName).value, (scalaBinaryVersion in artifactName).value)
+    //val resolvedArtifactName = artifactName.value(resolvedScalaVersion, projectID.value, artifact.value)
+    // if you would like to get a jar with version number embedded in it (as normally sbt does)
+    // uncomment the other definition of the `resolvedArtifactName`
+    val resolvedArtifact = artifact.value
+    val resolvedArtifactName = s"${resolvedArtifact.name}.${resolvedArtifact.extension}"
+    buildDirectory.value / "pack/lib" / resolvedArtifactName
+  },
+  copyrightString := "Copyright 2002-2013, LAMP/EPFL",
```

Comment:
Copied over from build.xml: https://github.com/scala/scala/blob/2.11.x/build.xml#L188
But yes, you are right.